[CI] Fixed esplora blockchain tests in CI

rajarshimaitra opened the pull request:
<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Description

Fixes #431 and esplora blockchain test skips in CI

### Notes to the reviewers


### Checklists

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

#### Bugfixes:

* [x] I'm linking the issue being fixed by this PR

notmandatory commented on Cargo.toml:
```diff
@@ -24,7 +24,7 @@ rand = "^0.7"
 electrum-client = { version = "0.8", optional = true }
 reqwest = { version = "0.11", optional = true, features = ["json"] }
-ureq = { version = "2.1", default-features = false, features = ["json"], optional = true }
+ureq = { version = "2.1", default-features = false, features = ["json", "tls"], optional = true }
```
Since `tls` the only default feature for ureq I'd change this line to:
```suggestion
ureq = { version = "2.1", features = ["json"], optional = true }
```
Also per below comment should be fine to have `tls` always enabled:
https://github.com/bitcoindevkit/bdk-cli/pull/41#discussion_r698333498

rajarshimaitra replied:
Noted. Will be done one a different PR.

notmandatory reviewed (changes requested):
Looks good, also checkout related https://github.com/tcharding/bdk/pull/2. We should be able to test esplora with ureq and reqwest. If you can incorporate into this PR I'll close the one I did.

rajarshimaitra committed:
Update Cargo.toml

- Changed to local bdk-macro
- Added back tokio
- Update esplora-reqwest and test-esplora feature guards

rajarshimaitra committed:
Fix reqwest blockchain test

- add back await_or_block! to bdk-macros
- use await_or_block! in reqwest tests

rajarshimaitra committed:
Test refactor

- Fix esplora module level feature flag
- Move esplora blockchain tests to module, to cover for both variants

rajarshimaitra commented:
Updated with suggested refactors and async changes. 

Rearranged the commits.

Moved out `tls` changes.

rajarshimaitra committed:
Update CI blockchain tests

notmandatory commented:
ACK 2caa590438205b3d9a60c46cabf90785d5b08069

tcharding commented on .github/workflows/cont_integration.yml:
```diff
@@ -78,15 +78,20 @@ jobs:
           - name: esplora
+            features: test-esplora,use-esplora-reqwest
+          - name: esplora
+            features: test-esplora,use-esplora-ureq    
```
This line has trailing whitespace.

tcharding commented on .github/workflows/cont_integration.yml:
```diff
@@ -104,8 +109,8 @@ jobs:
-        run: cargo test --features test-${{ matrix.blockchain.name }} ${{ matrix.blockchain.name }}::bdk_blockchain_tests
-
+        run: cargo test --no-default-features --features ${{ matrix.blockchain.features }} ${{ matrix.blockchain.name }}::bdk_blockchain_tests
+  
```
This line has trailing whitespace.

tcharding commented on Cargo.toml:
```diff
@@ -38,6 +38,10 @@ bitcoinconsensus = { version = "0.19.0-3", optional = true }
 
+# Platform-specific dependencies
+[target.'cfg(not(target_arch = "wasm32"))'.dependencies]
+tokio = { version = "1", features = ["rt"] }
```
Please correct me if I'm wrong, but this PR re-introduces a hard dependency on `tokio` just so we can test `esplora-reqwest` feature, that seems like a bad choice to me. A user of `bdk` that wishes to use `reqwest` will always be using `async`, why do we want to introduce a hard dependency on `tokio` just to be able to test in a manner that the library will not be used?

The `esplora-reqwest` feature can be tested coupled with either `async-interface` feature or using WASM target.

tcharding commented on src/blockchain/esplora/reqwest.rs:
```diff
@@ -106,19 +106,19 @@ impl Blockchain for EsploraBlockchain {
 
     fn get_tx(&self, txid: &Txid) -> Result<Option<Transaction>, Error> {
-        Ok(self.url_client._get_tx(txid).await?)
+        Ok(await_or_block!(self.url_client._get_tx(txid))?)
```
I do not like this usage of `await_or_block` because there is no reason to use `reqwest` in a blocking environment, we have `ureq` for that.

tcharding reviewed (commented):
Thanks for looking into this. I'm glad to see the things I missed in CI being remedied, however I'm not super fond of the re-introduction of hard dependency on `tokio` - seems like a step backwards to me.

tcharding commented:
All my comments are resolved by the implementation in #433, please see if you have any concerns with that approach @rajarshimaitra. Thanks.

notmandatory commented:
@tcharding @LLFourn as @rajarshimaitra noted on #433 even if this PR isn't the final solution it will unblock some other work (https://github.com/bitcoindevkit/bdk-cli/pull/41 and #429).  Also it's more of a roll back to how the `esplora::bdk_blockchain_tests` used to work before `ureq` was introduced.  Do you guys have any objection to merging this now while you work on a cleaner solution?

tcharding commented:
> @tcharding @LLFourn as @rajarshimaitra noted on #433 even if this PR isn't the final solution it will unblock some other work ([bitcoindevkit/bdk-cli#41](https://github.com/bitcoindevkit/bdk-cli/pull/41) and #429). Also it's more of a roll back to how the `esplora::bdk_blockchain_tests` used to work before `ureq` was introduced. Do you guys have any objection to merging this now while you work on a cleaner solution?

Don't stop the progress on my account :)
